### ISSUE TITLE
`export default` a function without parentheses causes "EmptyStatement"

### ISSUE THREAD

**Issue by zhanzhenzhen** [eslint/eslint#14057]
**Tell us about your environment**

* **ESLint Version:7.18.0
* **Node Version:14.15.4
* **npm Version:6.14.10

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "root": true,
    "parserOptions": {
        "ecmaVersion": 11,
        "sourceType": "module",
        "ecmaFeatures": {
        }
    },
    "rules": {
        "no-restricted-syntax": ["error", "EmptyStatement"]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Test 4 pieces of code:

```js
export default function() {return true};
```

```js
export default (function() {return true});
```

```js
export default function() {return true}
```

```js
let a = function() {return true};
```

**What did you expect to happen?**

1. No error
2. No error
3. No error
4. No error

**What actually happened? Please include the actual, raw output from ESLint.**

The first one reports an error (EmptyStatement):

```js
export default function() {return true};
```

```
  1:40  error  Using 'EmptyStatement' is not allowed  no-restricted-syntax
```

The other 3 works correctly.

I use this rule to prevent these styles of code:

```js
if (condition);
for (let i = 0; i < 3; i++);
```

This works fine for all of my code, except this `export default`, which is parsed as EmptyStatement but looks like it isn't.

I don't know if it's a bug of esprima, or it's an intended behavior.

**Comment by zhanzhenzhen** [comment]
It seems that I'm wrong. It's not a bug. The form `export default function` doesn't contain a semicolon. See:
https://tc39.es/ecma262/#table-export-forms-mapping-to-exportentry-records
